### DISCUSSION TITLE
Filtering/counting results

### DISCUSSION THREAD

**Discussion by coltspackers** [grafana/grafana-json-datasource#368] [Q&A]
Is anyone aware of a way to filter and count results from a panel that's using this plugin?

Basically, my panel is returning JSONata fields, and I want to count the number of fields returned that contain a certain string. So if I return 6 fields and only 3 contain the string, I want the panel to just show a "3".

I think there should be a way to do this via panel settings or transformations, but for some reason i'm really struggling to get it to work.

**Comment by zoltanbedi** [comment]
I think what you are looking for is the reduce transformation. See https://grafana.com/docs/grafana/latest/panels-visualizations/query-transform-data/transform-data/?utm_source=grafana#reduce

**Reply by coltspackers** [reply]
Unfortunately that's not working. I am returning strings from my JSON object, and am trying to use Value Mappings in the panel to convert one kind of string to a "1" and any other string to a "0", and then use the Reduce transform to sum up all the "1"s. However, the Reduce transform instead just converts the 1s and 0s back to their strings and doesn't do any reducing at all.

**Reply by coltspackers** [reply]
It seems that the Reduce transform is not actually getting those 1s and 0s to work with; it's getting the original strings from my JSON object.

**Reply by coltspackers** [reply]
Even leaving as a panel of 6 results of 0s and 1s, and then creating another panel to reference this 1s and 0s panel, is running into the same issue (where the new panel is not able to combine all 6 results into a sum total, because it doesn't actually see them as 1s and 0s)
